gateway/assets/script/bootstrap.js

```diff
@@ -320,8 +320,9 @@ app.controller("home", ['$scope', '$log', '$http', '$location', '$interval', '$f
 ]);
 
 function uuidv4() {
+    var cryptoInstance = window.msCrypto ? window.msCrypto : crypto;
```

Comment:
This was my input.. @kenfdev does it look OK to you?

Reply:
This should also work

```
var cryptoInstance = window.msCrypto || crypto;
```

Reply:
Both would work but I'd go with 

```
var cryptoObj = window.crypto || window.msCrypto; // for IE 11
```

as the MDN docs says.

https://developer.mozilla.org/en-US/docs/Web/API/Window/crypto

Reply:
I tried that approach and it failed to work in IE11 giving an exception, this is how we ended up with the above.

Reply:
@alexellis I did a test with the latest changes on all 4 listed browsers&versions. What was the exception?